packages/tracking-sdk/src/main.ts

```diff
@@ -598,13 +610,65 @@ export default function main() {
 
     if (!flags) {
       warn(`failed to fetch feature flags, using fall-back flags`);
-      return fallbackFlags.reduce((acc, flag) => {
+      flags = fallbackFlags.reduce((acc, flag) => {
         acc[flag.key] = flag;
         return acc;
       }, {} as Flags);
     }
 
-    return flags;
+    return proxify(flags, rateLimited(FLAG_EVENTS_PER_MIN, sendCheckEvent));
+  }
+
+  function sendCheckEvent(_: keyof Flag, flag: Flag) {
+    void featureFlagEvent({
```

Comment:
This can cause unhandled promise rejections which can look scary. Let’s catch them and log a warning or similar.